test.py

```diff
@@ -0,0 +1,162 @@
+# pyGene Fusion local testing
+import json
+from pydantic import BaseModel, validator, ValidationError, StrictStr, StrictInt, StrictBool
+from typing import Optional, List, Union
+
+class Gene(BaseModel):
+    type = 'GeneDescriptor'
+    value_id: str
+    label: StrictStr
+
+    @validator('value_id')
+    def is_valid(cls,v):
+        assert v.find(':') != -1 and v.find(' ') == -1, 'id must be a CURIE'
+        return v
+
+class SequenceRegion(BaseModel):
+    type = 'SequenceLocation'
+    seq_id: str
+    start: StrictInt
+    end: StrictInt
+
+    @validator('seq_id')
+    def is_curie(cls, v):
+        assert v.find(':') != -1 and v.find(' ') == -1, 'chr must be a CURIE'
+        return v
+
+class ChromosomeRegion(BaseModel):
+    type = 'ChromosomeLocation'
+    chr: str
+    start: str
+    end: str
+
+    @validator('chr')
+    def valid_chr(cls, v):
+        assert v.isalnum(), 'chr must have only alphanumeric characters'
+        return v
+
+    @validator('start', 'end')
+    def valid_loc(cls, v):
+        assert v[0] == 'p' or v[0] == 'q', 'start/end must start with p or q'
+        return v
+
+class GenomicRegion(BaseModel):
+    type = 'computed'
+    description: Optional[str] = None
+    value: Union[SequenceRegion, ChromosomeRegion]
+
+class TranscriptRegion(BaseModel):
+    component_type = 'transcript_segment'
+    transcript: str
+    exon_start: StrictInt
+    exon_start_offset : StrictInt
+    exon_end: StrictInt
+    exon_end_offset : StrictInt
+    gene: Gene
+    component_genomic_region: GenomicRegion
+
+    @validator('transcript')
+    def correct_form(cls, v):
+        assert v.find(':') != -1 and v.find(' ') == -1, 'id must be a CURIE'
+        return v
+
+class CriticalDomain(BaseModel):
+    status: StrictStr
+    name: str
+    id: str
+    gene: Gene
+
+    @validator('status')
+    def correct_status(cls, v):
+        assert v == 'lost' or v == 'preserved', 'status must be either lost or preserved'
+        return v
+
+    @validator('id')
+    def is_valid(cls, v):
+        assert v.find(':') != -1 and v.find(' ') == -1, 'id must be a CURIE'
+        return v
+
+class Event(BaseModel):
+    event_type: str
+
+    @validator('event_type')
+    def event_validate(cls, v):
+        assert v == 'rearrangement' or v == 'read-through' or v == 'trans-splicing', \
+            'event entry must be one of rearrangement, read-through, or trans-splicing'
+        return v
+
+class Linker(BaseModel):
+    linker_sequence: str
+
+    @validator('linker_sequence')
+    def valid_input(cls, v):
+        assert set(v) <= set('ACGT'), 'Linker sequence must only contain A,C,G,T'
```

Comment:
Should we allow users to input lower case? If so:
```suggestion
        assert set(v.upper()) <= set('ACGT'), 'Linker sequence must only contain A,C,G,T'
```

Reply:
Editing validator to convert to upper case.